### DISCUSSION TITLE
Is it possible to develop regular games other than virtual reality?

### DISCUSSION THREAD

**Discussion by WOLFxxxxxx** [bjornbytes/lovr#812] [General]
Why does it only specialize in virtual reality games? Will the library in the future be interested in all types of games like love2d?

**Comment by immortalx74** [comment]
It's perfectly capable of developing any kind of game. You can totally ignore the VR part and make a 3D/2D game, and In fact many people use it this way. If you're coming from love2d you'll be right at home, as the API is very similar.

This just my opinion from a user's perspective

**Reply by WOLFxxxxxx** [reply]
How do I disable virtual reality on Android devices
And get rid of the black screen when building and running the game

**Comment by bjornbytes** [comment]
> Why does it only specialize in virtual reality games?

I made LÖVR way back in 2016 because I wanted a quick way to get started with VR development and there weren't (m)any easy, open source options for that.  I still mostly work on VR projects and LÖVR's niche focus means I don't have to feel pressured to work on areas that aren't useful for VR like fancy desktop window stuff or mobile phone support.

> Will the library in the future be interested in all types of games like love2d?

LÖVE isn't interested in supporting all types of games, it's only interested in 2D games for PC/mobile.  With some work you can use it to make 3D games, but it's not the main focus.  LÖVR is the same way.  It's possible to disable VR and make flatscreen 3D games or 2D games with it, but its main focus is for making 3D VR games.  I'm not sure if that will change in the future, LÖVR's development process is pretty fluid and depends on requests/contributions from the community.

**Reply by WOLFxxxxxx** [reply]
How do I disable virtual reality on Android devices
And get rid of the black screen when building and running the game

**Reply by bjornbytes** [reply]
Hey, this isn't very easy to do right now out of the box, but people have done it and there are some patches floating around in the Matrix/Discord that will render a project on an Android phone.

**Reply by WOLFxxxxxx** [reply]
Can I get a link to a project, article, or publication that talks about this topic?

**Reply by bjornbytes** [reply]
Here's the most recent patch, it was actually updated just last week so it should be in a pretty good state:

https://github.com/bjornbytes/lovr/compare/dev...jmiskovic:lovr:phone-lovr

and here's the last time it was discussed in chat

https://matrix.to/#/!XVAslexgYDYQnYnZBP:matrix.org/$17252934133076TZQoj:lovr.org?via=lovr.org&via=matrix.org&via=nitro.chat

**Reply by WOLFxxxxxx** [reply]
Thank you very much ♥️

**Reply by jmiskovic** [reply]
It doesn't really work out of the box, as you need to compile `libVkLayer_khronos_synchronization2.so` and insert it into build process with the rest of libs. After that the lovr will run on the phone if you have `modules.headset = false` in conf.lua file.

Touches are supported, gyro and accelerometer are available in Lua. Portrait/landscape orientations work, can be rotated on the fly (but it restarts the app). You can bundle your project with APK, you can also put it on the external storage for the app to dynamically load it from there.

The biggest next issue to solve is suspending and resuming the app. On resuming the app either crashes or just has black screen.

Feel free to open a ticket on my fork if you need help with what's currently there, but also please don't expect more than a proof of concept.

**Reply by WOLFxxxxxx** [reply]
Thank you very much ♥️
